### PR TITLE
fix(query-plans): stop cost/cardinality estimates leaking into plan fingerprints

### PR DESCRIPTION
**Pull request by joeharris76**
## Summary

The plan fingerprint is documented as a **logical, stats-independent** structural hash — a `VACUUM`/`ANALYZE` or row-count change must not change it. Two parsers violated that by folding estimate-bearing EXPLAIN detail into signature-bearing fields:

- **DuckDB** FORMAT JSON `extra_info` is a dict carrying `Estimated Cardinality`, folded into `join_conditions`/`aggregation_functions`/`filter_expressions`/`sort_keys`. (Confirmed live: a GROUP BY hashed differently at 200 vs 9000 rows.)
- **DataFusion** EXPLAIN ANALYZE appends a `metrics=[output_rows=..]` block that the filter detail captured.

PostgreSQL and Redshift route cost/rows into dedicated, non-hashed `properties`, so they were already clean; all other recorded fixtures were verified clean.

### Changes
- Added shared `strip_estimates(text)` / `strip_estimate_keys(mapping)` helpers in `parsers/base.py` — the single place the estimate-stripping rule lives.
- DuckDB strips estimate keys from the `extra_info` dict (raw kept in the non-hashed `physical_operator.platform_metadata`); DataFusion strips the `metrics=[...]` block from the detail before structural extraction.
- The strip is anchored to **unambiguous** estimate wording (`Estimated Cardinality`, `EC:`, `metrics=[...]`, `(cost=N..N rows=N width=N)`) so a genuine predicate on a column literally named `cost`/`rows`/`num_rows`/`cardinality`/`ec` is never corrupted.
- New `tests/unit/query_plans/test_fingerprint_hygiene.py`: registry-wide invariant asserting (a) no parser leaks an estimate token into the signature, with a coverage guard so a new parser must add a fixture, and (b) estimate-only-different fixtures hash identically (DuckDB/DataFusion exercise the strip path; PostgreSQL/Redshift assert the complementary properties-path guarantee).
- Live DuckDB row-count stability regression test in `test_duckdb_plan_capture.py`.
- Updated the `query_plan_models.py` stability contract to drop the per-engine stats-independence caveat now that the invariant enforces it.

## Testing
- `uv run -- python -m pytest tests/unit/core/query_plans/ tests/unit/query_plans/ tests/unit/platforms/test_duckdb_plan_capture.py -q` → **576 passed**
- `ruff check` / `ruff format --check` / `ty check` → all green
- `make pr-preflight`: lint/format/type green and all query-plan suites pass. The 19 failing tests are pre-existing **sandbox-environmental** (worktree pool absent, delta/ducklake extensions missing, root-writable-dir CLI tests, git changelog/squash) — none import or touch the changed files.

## Deviations from the TODO
- The TODO suggested `strip_estimates(text)`; I added a second `strip_estimate_keys(mapping)` helper because DuckDB's `extra_info` is a structured dict (key-based stripping is cleaner and safer than regex over JSON).
- **Known impact:** fingerprint *values* for DuckDB/DataFusion JOIN/FILTER/AGGREGATE/SORT plans change vs pre-fix captures — this is the intended correction and is consistent with the documented "not comparable across engine versions" property; stored history/baseline fingerprints recompute on the next run.

Closes `query-plan-capture-fingerprint-estimate-leak` (moved to `_project/DONE/`).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_018fT8yfjtBSL1jUZUQArwvx)_

### CODE REVIEW

File: benchbox/core/query_plans/parsers/base.py [hunk 1]
```diff
@@ -25,6 +26,136 @@
...
+    ``num_rows`` (e.g. ``(cost=5 AND x=1)`` or ``num_rows = 5``) is left intact.
+    Returns the cleaned, whitespace-normalized string.
+    """
+    if not text:
+        return text
+    cleaned = _METRICS_BLOCK_RE.sub("", text)
+    cleaned = _COST_PAREN_RE.sub("", cleaned)
+    cleaned = _INLINE_ESTIMATE_RE.sub("", cleaned)
+    # Tidy separators left behind by removals (trailing commas, empty brackets).
+    cleaned = re.sub(r"\[\s*\]", "", cleaned)
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply benchbox/core/query_plans/parsers/base.py:152]: Fixed in #829 — `strip_estimates` only runs the empty-bracket / trailing-comma tidy-up when an estimate token was actually removed, so a genuine empty-list predicate (`arr = []`) is preserved in DuckDB/DataFusion signatures. Regression test added.

---
_Generated by [Claude Code](https://claude.ai/code)_
